browserdebuggertools/eventhandlers.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from abc import ABCMeta, abstractmethod
+
+from browserdebuggertools.exceptions import DomainNotEnabledError
+
+logging.basicConfig(format='%(levelname)s:%(message)s')
+
+
+class EventHandler(object):
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def handle(self, message):
+        pass
+
+
+class PageLoadEventHandler(EventHandler):
+
+    def __init__(self, socket_handler):
+        super(PageLoadEventHandler, self).__init__()
+        self._socket_handler = socket_handler
+        self._url = None
+        self._root_node_id = None
+
+    def handle(self, message):
+        super(PageLoadEventHandler, self).handle(message)
+        if message.get("method") == "Page.navigatedWithinDocument":
+            logging.info("Detected URL change %s" % message["params"]["url"])
+            self._url = message["params"]["url"]
+        elif message.get("method") == "Page.domContentEventFired":
+            logging.info("Detected Page Load")
+            self._reset()
+
+    def _reset(self):
+        self._url = None
+        self._root_node_id = None
+
+    def check_page_load(self):
+        try:
+            self._socket_handler.get_events("Page")
```

Comment:
It is a bit confusing that on the one hand, "Page" is an 'internal event' in the socket handler, but on the other hand we expose it